netx/resolver/dnsoverhttps.go

```diff
@@ -14,14 +14,15 @@ import (
 // DNSOverHTTPS is a DNS over HTTPS RoundTripper. Requests are submitted over
 // an HTTP/HTTPS channel provided by URL using the Do function.
 type DNSOverHTTPS struct {
-	Do  func(req *http.Request) (*http.Response, error)
-	URL string
+	Do           func(req *http.Request) (*http.Response, error)
+	URL          string
+	HostOverride string
 }
 
 // NewDNSOverHTTPS creates a new DNSOverHTTP instance from the
 // specified http.Client and URL, as a convenience.
-func NewDNSOverHTTPS(client *http.Client, URL string) DNSOverHTTPS {
-	return DNSOverHTTPS{Do: client.Do, URL: URL}
+func NewDNSOverHTTPS(client *http.Client, URL, HostOverride string) DNSOverHTTPS {
+	return DNSOverHTTPS{Do: client.Do, URL: URL, HostOverride: HostOverride}
```

Comment:
Like before, I suggest introducing an extra factory so that we only see the two extra arguments when they are mostly needed, and otherwise we keep existing code untouched. Should lead to something like:

```suggestion
func NewDNSOverHTTPS(client *http.Client, URL string) DNSOverHTTPS {
	return DNSOverHTTPSWithHostOverride(client, URL)
}

// NewDNSOverHTTPSWithHostOverride is like NewDNSOverHTTPS except that
// it's creating a resolver where we use the specified host.
func NewDNSOverHTTPSWithHostOverride(client *http.Client, URL, hostOverride string) DNSOverHTTPS {
	return DNSOverHTTPS{Do: client.Do, URL: URL, HostOverride: hostOverride}
```